### ISSUE TITLE
Rage (9200)

### ISSUE THREAD

**Issue by LeJimster** [ValveSoftware/Proton#166]
It appears all the major bugs for this game have now been fixed.  I plan to play through the game before I recommend whitelisting it.

Old bugs:
~~Rage 32-bit suffers from Virtual Texture bug.~~
~~Rage 64-bit renders correctly, although needs workaround to enable audio~~ * Audio fixed as of Proton 3.16-5
Mesa bug reports:
~~[Hang/Freeze in menus](https://bugs.freedesktop.org/show_bug.cgi?id=107891)~~  *[Fixed in latest mesa-git](https://cgit.freedesktop.org/mesa/mesa/commit/?id=7a992fcfa059b0db0f07bc3eb38050eaab7aaf34)
~~[MegaTexture corruption](https://bugs.freedesktop.org/show_bug.cgi?id=107694)~~  *[Fixed in latest mesa-git as of 24/6/22](https://gitlab.freedesktop.org/mesa/mesa/-/commit/179ad03f5d56ce5b84f81edbe7ca3af752cdac18)

[Steam log](https://github.com/ValveSoftware/Proton/files/2354352/steam-9200.zip)
[My System Specs](https://gist.github.com/LeJimster/7751f48f10bf0082158bace47dec7430)

32/64 video comparison showing the Virtual Texture (aka MegaTexture) problem.
[![Rage 32 mesa megatexture corruption](https://img.youtube.com/vi/EUwq4dbnFkA/0.jpg)](https://www.youtube.com/watch?v=EUwq4dbnFkA)

[**Performance tweaks**](http://rage.wikia.com/wiki/Tweaks)
From my own experimenting I found these _Launch Options_ produce the smoothest overall framerate on my system (RX Vega 56/Ryzen 1700)
`+r_swapInterval 1 +jobs_numThreads 2 +vt_maxPPF 64 +g_fov 90`

It seems like higher cpu thread usage than 2 offers little benefit while causing the frame rate to dip and stutter in certain areas.  I also chose to set the fov to 90 as 80 feels too low gaming at 1440p.

**Comment by tr37ion** [comment]
Arch Linux
Kernel 4.18.3
Proton 3.7
Nvidia GTX 1070 - nvidia-beta 396.54
KDE Plasma 5.13.4
KDE Framework 5.49.0
Qt 5.11.1
§ python --version = 3.7.0

Starting Rage (32Bit) the first time it took 2-3min until the game did start (Megatextures loading?)
I set all GFX settings to MAX and did try to start the campaign. The game crashed.

I started the game again (32Bit) and now everything seems to work perfectly(!). Playing the campaign for some minutes was fine.

Candidate for [Whitelist] ?

**Comment by vpelletier** [comment]
Played from start to just before "mutant bash tv" mission, including pyros cave & refinery. The only issue I had was sound becomming distorted (a lot of cracks and pops - maybe a sound buffer being written to while it is being read by hardware ?). Onset may take time, as sound is initially fine. Only tested the default launch option (so... 32bits I guess ? the other option being "64-bits with mods", not sure whether the difference is "64 bits", "mods", or both).

Distribution: Debian sid
Kernel: 4.17.17 (package 4.17.0-3-amd64)
Proton: 3.7-4
CPU: i7-6700
GPU: GTX 1060@6GB, driver 396.51-1 (from "experimental" package repository)
pulseaudio 12.0-1, sink alsa_output.pci-0000_01_00.1.hdmi-stereo  (HDMI0 on nvidia card)

**Comment by Kira-PH** [comment]
I'm getting a hang when trying to load Hagar Caves, which comes after the first mission that requires loading a different area (so it's not just an area loading issue)

Update: Also hangs on loading the Wasted Garage

Update: Installed Mesa 18.2 and LLVM 7 and this issue has gone away

Update: Removed Mesa 18.2 (because I have an NVidia card), purged & reinstalled LLVM 7 and this issue came back. Reinstalled Mesa 18.2 and it didn't go away again.

**Comment by whizse** [comment]
Hangs in the menu every time for me. No crash no errors, if I'm fast enough I can load a game. In-game everything including the menu is fine. Esync on or off makes no difference. 

@LeJimster I filed a bug about the texture artifacts, it seems to be radeonsi specific. There's a suggested workaround there too. Check if it works for you:
https://bugs.freedesktop.org/show_bug.cgi?id=107694

**Comment by Kira-PH** [comment]
After the most recent update RAGE was starting with a black screen. Deleting its compatdata folder meant it was reinitialised with a clean wineprefix, but now it complains about being out of memory and ultimately loses a significant proportion of its textures (making it fairly difficult to play...)

This does mean there may be issues with the upgrade from one Proton version to the next.

**Comment by Kira-PH** [comment]
Still unable to get into The Wasted Garage but this time I was able to enter Hagar Caves - although I may have done so earlier, so this could be somehow related to a mission event making a global change that affects the game.

**Comment by LeJimster** [comment]
UPDATE:
I've found a fix for missing sound in the 64-bit version.
Installing `xact `to the Rage prefix did the trick.  I used this command:
`WINEPREFIX=/your path to/Steam/SteamLibrary/steamapps/compatdata/9200/pfx winetricks xact
`
Played the 64-bit version for 2 or 3 hours tonight, no issues.

**Comment by tr37ion** [comment]
@LeJimster I can confirm installing `xact` fixes 64-Bit version audio.

While it seems that `xact` in general has issues with [DTS Patch](https://github.com/darealshinji/dcaenc) for Pulseaudio. Very choppy sound with my system.

**Comment by LeJimster** [comment]
@tr37ion seems like the broken audio is a much wider spread issue.  #54

For me the sound isn't choppy but it does appear like some of the sounds get cut or stop playing for a second here or there.  I notice it mostly on engine sounds.  Maybe that effect is exaggerated on your setup.

There is mention or Xaudio2 being broken on 64bit prefixes, I'm unfamiliar with these projects so I can't say what the best fix is.

**Comment by LeJimster** [comment]
@HughPH I have been playing the 64-bit fine but then it hanged when loading The Wasted Garage.  These issues were apparently fixed in the proprietary drivers long ago, but it seems the game is broken in mesa.  I found a strange workaround to get the level to load however.  Switch to windowed mode before entering The Wasted Garage and it loads fine.

**Comment by Kira-PH** [comment]
Thanks @LeJimster - I don't actually have Mesa, though (mine is an Nvidia card and I'm using the latest drivers) and I was playing the 32-bit version... But I will try it and see what happens.

**Comment by whizse** [comment]
@LeJimster are you sure it's Mesa issues? If you use native xaudio on 64bit you probably run into the same pointer problems with the xaudio dlls as mentioned in https://bugs.winehq.org/show_bug.cgi?id=38668

**Comment by lucifertdark** [comment]
using Winetricks/protontricks to set override for xaudio2_7 native, builtin is enough to get the sound back in 64bit. No crashes for me so far.

ps Installing xact is what's causing the crashes for most people, DON'T DO IT! ;)

update: looks like I spoke too soon, just after posting ^ I started up the 64bit game again & it crashed in under 5 minutes. The 32bit version is working just fine though.

**Comment by LeJimster** [comment]
@lucifertdark strange.  I haven't experienced any crashing on the 64bit version.  Although I only quickly tested the 32bit with the latest mesa-git.

What is your system specs? Hopefully its not another regression in mesa.  I will try to test later tonight.

**Comment by lucifertdark** [comment]
> @lucifertdark strange. I haven't experienced any crashing on the 64bit version. Although I only quickly tested the 32bit with the latest mesa-git.
> 
> What is your system specs? Hopefully its not another regression in mesa. I will try to test later tonight.

I should have added it earlier.. I'm an Nvidia user 

https://gist.github.com/lucifertdark/584c14275612e8e3c4a8070d428d8a02

**Comment by LeJimster** [comment]
I've just played 15 mins on the 64bit, no crashes on my RX Vega.  So maybe you broke your prefix, or there is a driver issue.  Would be nice to see if another nvidia user could test and compare.

**Comment by ziabice** [comment]
Game crashes on start complaining about a failed shader compilation: `Failed to compile render prog binkyuv from binkyuv`.

Using Proton 3.7-6.

I'm using Mesa 18.2.1 with LLVM 6.0.1.

The same problem happens if I use 32-bit or 64-bit version of RAGE.

[My Specs](https://gist.github.com/ziabice/8c19008e58f82b16a62f7ab65ba235d4)

[steam-9200.log](https://github.com/ValveSoftware/Proton/files/2407551/steam-9200.log)

**Comment by lucifertdark** [comment]
> Game crashes on start complaining about a failed shader compilation: `Failed to compile render prog binkyuv from binkyuv`.
> 
> Using Proton 3.7-6.
> 
> I'm using Mesa 18.2.1 with LLVM 6.0.1.
> 
> The same problem happens if I use 32-bit or 64-bit version of RAGE.
> 
> [My Specs](https://gist.github.com/ziabice/8c19008e58f82b16a62f7ab65ba235d4)
> 
> [steam-9200.log](https://github.com/ValveSoftware/Proton/files/2407551/steam-9200.log)

You need LLVM 7 or higher.

**Comment by LeJimster** [comment]
@ziabice you need mesa-git for Rage to work on mesa right now. (Should be fine from 18.3 release). Also updating LLVM is a good idea as it fixes crashing in other games.

**Comment by xDShot** [comment]
~~The game doesn't recognize my game controller, even when I enable it in game settings and configure it in Steam Big Picture for the game.~~

**Comment by ziabice** [comment]
@lucifertdark @LeJimster because Wolfenstein TNO works well with my current setup, I believed Rage had to work too, my bad! :/

**Comment by flibitijibibo** [comment]
@kisak-valve Requesting an XAudio2 label for this game...

**Comment by kwahoo2** [comment]
> Rage 32-bit suffers from Virtual Texture bug.
> Rage 64-bit renders correctly, although needs workaround to enable audio (see below)
> 
> Mesa bug reports:
> [Hang/Freeze in menus](https://bugs.freedesktop.org/show_bug.cgi?id=107891) *[Fixed in latest mesa-git](https://cgit.freedesktop.org/mesa/mesa/commit/?id=7a992fcfa059b0db0f07bc3eb38050eaab7aaf34)
> [MegaTexture corruption](https://bugs.freedesktop.org/show_bug.cgi?id=107694)
> 
> [Steam log](https://github.com/ValveSoftware/Proton/files/2354352/steam-9200.zip)
> [My System Specs](https://gist.github.com/LeJimster/7751f48f10bf0082158bace47dec7430)
> 
> 32/64 video comparison showing the Virtual Texture (aka MegaTexture) problem.
> [![Rage 32 mesa megatexture corruption](https://camo.githubusercontent.com/f8fa8522aff1e0289e7a7ada8e7a721ecec6a645/68747470733a2f2f696d672e796f75747562652e636f6d2f76692f455577713464626e466b412f302e6a7067)](https://www.youtube.com/watch?v=EUwq4dbnFkA)
> 


I can confirm both issues. The megatexture bug is a regression  - with initial release of SteamPlay Rage ran fine. My specs: https://gist.github.com/kwahoo2/41bf5ea9fe74de52269c6a349d3639cb

**Comment by oblitum** [comment]
Can this game run under Intel HD Graphics in Wine? I'm on a 7700K CPU alone, the game crashes on startup:

```
wine: Unhandled page fault on read access to 0x00000000 at address (nil) (thread 002a), starting debugger...
```

But I think this is relevant, since it tries to use cuda related stuff:

```
4307.491:0008:0009:trace:module:load_dll looking for L"cudart32_40_17.dll" in L"E:\\Steam\\steamapps\\common\\RAGE;C:\\Program Files (x86)\\Steam;C:\\windows\\system32;C:\\windows\\system;C:\\windows;.;C:\\windows\\system32;C:\\windows;C:\\windows\\system32\\wbem"
4307.491:0008:0009:trace:module:get_load_order looking for L"E:\\Steam\\steamapps\\common\\RAGE\\cudart32_40_17.dll"
4307.491:0008:0009:trace:module:get_load_order got hardcoded default for L"E:\\Steam\\steamapps\\common\\RAGE\\cudart32_40_17.dll"
4307.491:0008:0009:trace:module:load_builtin_dll Trying built-in L"E:\\Steam\\steamapps\\common\\RAGE\\cudart32_40_17.dll"
4307.491:0008:0009:warn:module:load_builtin_dll failed to load .so lib for builtin L"E:\\Steam\\steamapps\\common\\RAGE\\cudart32_40_17.dll": /mnt/seagate/Steam/steamapps/compatdata/9200/pfx/dosdevices/e:/Steam/steamapps/common/RAGE/cudart32_40_17.dll: invalid ELF header
4307.491:0008:0009:trace:module:load_builtin_dll Trying built-in L"cudart32_40_17.dll"
4307.491:0008:0009:warn:module:load_builtin_dll cannot open .so lib for builtin L"cudart32_40_17.dll": /mnt/seagate/Steam/steamapps/common/Proton 3.16/dist/lib/wine/cudart32_40_17.dll.so: cannot open shared object file: No such file or directory
4307.491:0008:0009:trace:module:load_native_dll Trying native dll L"E:\\Steam\\steamapps\\common\\RAGE\\cudart32_40_17.dll"
4307.492:0008:0009:trace:module:map_image mapped PE file at 0x330000-0x395000
4307.492:0008:0009:trace:module:map_image mapping section .text at 0x331000 off 400 size 41000 virt 40e4e flags 60000020
4307.492:0008:0009:trace:module:map_image mapping section .rdata at 0x372000 off 41400 size 12e00 virt 12cb2 flags 40000040
4307.492:0008:0009:trace:module:map_image clearing 0x384e00 - 0x385000
4307.492:0008:0009:trace:module:map_image mapping section .data at 0x385000 off 54200 size 1200 virt 237c flags c0000040
4307.492:0008:0009:trace:module:map_image clearing 0x386200 - 0x387000
4307.492:0008:0009:trace:module:map_image mapping section .rsrc at 0x388000 off 55400 size 6a00 virt 6838 flags 40000040
4307.492:0008:0009:trace:module:map_image clearing 0x38ea00 - 0x38f000
4307.492:0008:0009:trace:module:map_image mapping section .reloc at 0x38f000 off 5be00 size 6000 virt 5fc8 flags 42000040
4307.492:0008:0009:trace:module:perform_relocations relocating from 0x10000000-0x10065000 to 0x330000-0x395000
4307.492:0008:0009:warn:module:alloc_module disabling no-exec because of L"cudart32_40_17.dll"
4307.492:0008:0009:trace:module:set_security_cookie initializing security cookie 0x3851c0
```

There's just a `cudart32_40_17.dll` in the RAGE directory.

**Comment by LeJimster** [comment]
It shouldnt be cuda causing the crash as usually the game detects and disables GPU transcoding for non nvidia cards.  I haven't tested it on Intel but if you use mesa, you might need latest mesa-git.

**Comment by LeJimster** [comment]
> I can confirm both issues. The megatexture bug is a regression - with initial release of SteamPlay Rage ran fine. My specs: https://gist.github.com/kwahoo2/41bf5ea9fe74de52269c6a349d3639cb

Did you change your GPU since initial release of SteamPlay?  Because the bug has always existed for on AMD with mesa from my testing (and this is going back years).  Or was there something you changed with your setup between the intial release and now?  Drivers? Kernel? Distro? Were you using "radeon" or "amdgpu".  I wonder, because if you are correct..  You might hold the key to finding a fix.

Also, I experienced this bug from the very first release of proton...   It was one of the first, if not the first game I tried.

**Comment by oblitum** [comment]
@LeJimster I'm on mesa 18.2.4. Do you know what fix/commit mesa-git could have for this? Just curious because I can't use it on my main system.

**Comment by LeJimster** [comment]
@oblitum I'm not sure why its not working with 18.2 series.  I haven't tried them personally but have heard people say it doesn't work a few times now.

I believe the main change was this: https://patchwork.freedesktop.org/patch/244742/

And I read an article saying it had been added to 18.2.3.. 

Also a thought just occured to me, are you running steam-native or steam runtime?  Ive found in the past some games need steam runtime to launch, so I always use it now.

Just one more thing.  The guy who created a bug report for the mega texture bug in my first post. Tested against a Intel skylake GPU.  So I presume it *is* possible to play it with Intel.

**Comment by oblitum** [comment]
@LeJimster Yes I use steam-native because I had the reverse experience in the past, plain steam caused me issues on ArchLinux. I tried plain steam for this game but the effect was the same.

**Comment by LeJimster** [comment]
@oblitum I do have access to an Intel system, so might be able to do some testing for you, but not until Sunday at the earliest.

**Comment by oblitum** [comment]
@LeJimster oh, that would be nice, no rush. Just like to comment that the issue I've reported above doesn't seem related to shader compilation, which the freedesktop patch, and the previous discussion, seems to point to.

**Comment by oblitum** [comment]
Logs

[steam-9200.log](https://github.com/ValveSoftware/Proton/files/2544615/steam-9200.log)

The x64 seems to go a bit further, I get a message box stating there was a pure virtual function call and the game process ends, no game window is ever run. I can't send the log for this execution because it's 512MB.

**Comment by kwahoo2** [comment]
> > I can confirm both issues. The megatexture bug is a regression - with initial release of SteamPlay Rage ran fine. My specs: https://gist.github.com/kwahoo2/41bf5ea9fe74de52269c6a349d3639cb
> 
> Did you change your GPU since initial release of SteamPlay? Because the bug has always existed for on AMD with mesa from my testing (and this is going back years). Or was there something you changed with your setup between the intial release and now? Drivers? Kernel? Distro? Were you using "radeon" or "amdgpu". I wonder, because if you are correct.. You might hold the key to finding a fix.

Same GPU. The bug happened after one of Mesa or Steam updates. That was a working configuration IIRC https://gist.github.com/kwahoo2/e3c89a9b3798764991df3fc65bda4c2c

**Comment by ziabice** [comment]
With latest Proton 3.16-5 this game works perfectly. Just tested the 64-bit version and audio finally works like a charm. I haven't played too much, but to me, this game can be whitelisted. No workarounds of any sorts are needed, it installed and worked out of the box.

My specs: [https://gist.github.com/ziabice/8c19008e58f82b16a62f7ab65ba235d4#file-sysinfo-txt](https://gist.github.com/ziabice/8c19008e58f82b16a62f7ab65ba235d4#file-sysinfo-txt)

**Comment by oblitum** [comment]
Still [same error for me](https://github.com/ValveSoftware/Proton/issues/166#issuecomment-435559175) on 3.16-5.

**Comment by LeJimster** [comment]
As ziabice mentioned audio is now working on 64-bit out of the box with Proton 3.16-5 beta, presumably due to the great work Ethan Lee has been doing with FAudio.  For AMD users it's very playable with the 64-bit, but would be nice to have the 32-bit rendering correctly.

@oblitum I forgot to update you, I tested the game on an Intel system and was getting the same errors as yourself.  There is a new **Intel Iris Gallium3D driver** thats experimental right now, might be worth a test.

**Comment by oblitum** [comment]
@LeJimster OK thanks, no problem.

**Comment by rado84-github** [comment]
Proton 4.2-2, changing brightness level in game has no effect and in dark areas it's impossible to find the exit.
Everything else works great.

**Comment by LeJimster** [comment]
Can confirm the brightness change doesn't work on 32-bit, but does on 64-bit.  I checked on windows and brightness control works on both versions fine.  My setup/system is in the original post (AMD Vega 56/Ryzen 7), I checked this on 4.2-2, 3.16-8, 3.7-8.

Would be nice to know your system @IvoAndonov as it might be driver specific and not a proton bug.

**Comment by rado84-github** [comment]
> Can confirm the brightness change doesn't work on 32-bit, but does on 64-bit. I checked on windows and brightness control works on both versions fine. My setup/system is in the original post (AMD Vega 56/Ryzen 7), I checked this on 4.2-2, 3.16-8, 3.7-8.
> 
> Would be nice to know your system @IvoAndonov as it might be driver specific and not a proton bug.

My system is 64-bit and it doesn't work.

Mint 18.3 Cinnamon x64 with kernel 4.13.0-45, Proton 4.2-2, driver version 418.56 and here's a copy-paste of my hardware from a forum signature: AsRock B85M Pro4, Core i3-4170, Palit GTX 1050 Ti 4 GB, Corsair 32 GB DDR3 1600 MHz, Corsair CX750, Corsair Force LE 120 GB, Western Digital 1.5 TB

**Comment by LeJimster** [comment]
I'm talking about the game itself, there are 2 executables for Rage.

![64-bit_Rage](https://user-images.githubusercontent.com/35663469/56379594-f8349700-6207-11e9-9a19-162ced150ef9.jpg)

**Comment by LeJimster** [comment]
@IvoAndonov I'm talking about the game itself, there are 2 executables for Rage.

![64-bit_Rage](https://user-images.githubusercontent.com/35663469/56379594-f8349700-6207-11e9-9a19-162ced150ef9.jpg)

**Comment by rado84-github** [comment]
I tried the 64-bit too. Brightness does work there but that version says it doesn't support 1920x1080 fullscreen and it remains in a window part of which is behind the panel. Also the cursor is limited and I can't turn the charater in 180 degrees, let alone 360.

**Comment by dreamer** [comment]
Regarding brightness - can you test this workaround: https://github.com/ValveSoftware/Proton/issues/1550#issuecomment-487287891 ?

**Comment by phrogg** [comment]
I'm still having the same issue. How did you guys fixed this?

I have two RX480. And Mesa version 19.0.1 padoka.

**Comment by phrogg** [comment]
I upgraded to MESA 19.2.0 devel still the same issue.

**Comment by dreamer** [comment]
@LuigiTheHunter what issue?

**Comment by LeJimster** [comment]
> Regarding brightness - can you test this workaround: [#1550 (comment)](https://github.com/ValveSoftware/Proton/issues/1550#issuecomment-487287891) ?

That registry entry seems to give some control over brightness, but compared to the 64-bit version it doesn't get quite as light.

> I upgraded to MESA 19.2.0 devel still the same issue.

Rage works fine with the 64-bit version of the game on mesa.  There are launch options that might help with performance in my original post.  If you are experiencing some other issue, please give details.

**Comment by phrogg** [comment]
@LeJimster I still have the same issues, from the first thread. I will try to use the tweaking options from there today. Also I will send a screenshot from my issue.

I still have these issues:
![20190522091102_1](https://user-images.githubusercontent.com/1367949/58156487-773d4500-7c76-11e9-8569-9d773f597338.jpg)
![20190522091109_1](https://user-images.githubusercontent.com/1367949/58156488-773d4500-7c76-11e9-992e-b2c00febe06b.jpg)
![20190522091116_1](https://user-images.githubusercontent.com/1367949/58156489-773d4500-7c76-11e9-8b7a-91b9046349a7.jpg)

**Comment by kopr12** [comment]
I have texture problems both in 32bit and 64bit
They are far worse than in some pictures posted here, you can barely recognize anything.
mesa 19.1.1
amdgpu driver
Proton 4.2-9

![20190630-115615](https://user-images.githubusercontent.com/11412353/60395082-5f3aca00-9b2e-11e9-8274-d3ab597193c4.png)

**Comment by phrogg** [comment]
@kajzersoze I have the same problem still.

**Comment by dreamer** [comment]
(I don't own the game, just trying to help here) Have you tried using [`r_useSMP "1"`](https://rage.fandom.com/wiki/Tweaks) option?

Otherwise, according to PCGW this game uses exclusively OpenGL, so these rendering bugs should be directed rather to the [Mesa bugtracker](https://bugs.freedesktop.org/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&bug_status=REOPENED&component=Drivers%2FGallium%2Fradeonsi&product=Mesa).

**Comment by kopr12** [comment]
There's this problem on Windows too, AMD cards only as well, seems to be solvable in AMD Control Center, by turning on 'Surface Format Optimization'

[How to fix black squares in textures (AMD problem only)](https://steamcommunity.com/sharedfiles/filedetails/?id=313663882)

**Comment by phrogg** [comment]
@kajzersoze ahh cool found!

**Comment by NeoTheFox** [comment]
I'm getting 
`Failed to compile render prog binkyuv from binkyuv` [Full log](https://gist.github.com/NeoTheFox/b07329a877d91d09db9a8adec2fb09de)
and my LLVM is 9.0.0, so maybe a regression? I'm running ArchLinux with Vega 56 + Mesa on Ryzen 1600X

**Comment by DianaNites** [comment]
Any way to get the same fix for windows for proton?

**Comment by Narann** [comment]
> turning on 'Surface Format Optimization'

How can this be achieved with mesa amd Proton ? (AMD RX580)

**Comment by phrogg** [comment]
I don't know if this is proton related or graphics card related but for me the 64 bit version works now without any issues at all. However I would like to get the achievements so ....

I also noticed i have these startup options enabled: `PROTON_NO_ESYNC=1 %command% +cvaradd g_fov 12 +com_skipIntroVideo 1 +image_anisotropy 16 +image_use +r_swapInterval 1 +jobs_numThreads 2 +vt_maxPPF 64 +g_fov 90`

**Comment by kopr12** [comment]
64 bit version seems to be better now but still bad.
32 bit is the same.

**Comment by kopr12** [comment]
Did some more testing and the Proton version and mesa don't matter, what matters is this option vt_maxPPF , setting that one to 128 makes this problem less visible in my case, without it it's horrid, tried few more settings (not all) but those didn't make a difference.
Using only this
`%command% +com_skipIntroVideo 1 +vt_maxPPF 128`

**Comment by phrogg** [comment]
You are right. I tested to confirm: `+vt_maxPPF 1` makes it really terrible and `+vt_maxPPF 128` does make it nearly playable for 32bit at least on my machiene. 64bit however now works really good.

**Comment by kopr12** [comment]
32bit and 64bit are the same for me in regard to this problem.
No more no less.
It's somewhat playable with `+vt_maxPPF 128` , compared to how it was without that option, but the textures are still corrupt and it's very annoying.
That's it I guess, for now, I feel that the problem could be solved in Mesa like that option in Windows driver mentioned before.

**Comment by RussianNeuroMancer** [comment]
Is anyone can provide apitrace to mesa dev?

https://gitlab.freedesktop.org/mesa/mesa/-/issues/1326#note_471422

**Comment by allanpfreitas** [comment]
Hi everybody, this error persist, any clue ?

***************************
Failed to compile render prog binkyuv from binkyuv
***************************`

**Comment by allanpfreitas** [comment]
Guys how to run this game using Proton ? I'm using Ubuntu 20.04, any help is appreciated

Failed to compile render prog binkyuv from binkyuv

**Comment by LeJimster** [comment]
@allanpfreitas Hard to help without more information. 

> 3. You can retrieve a full system information report by clicking
>    `Help` > `System Information` in the Steam client on your machine.
> 4. Please copy it to your clipboard by pressing `Ctrl+A` and then `Ctrl+C`.
>    Then paste it in a [Gist](https://gist.github.com/) and post the link in
>    this issue.

Nvidia and AMD GPU users should be able to get the game working, haven't checked Intel for some time.

**Comment by allanpfreitas** [comment]
Hi @LeJimster thanks for your reply. Here what you requested https://gist.github.com/allanpfreitas/73e5b787491b20286cea2dc1841e037e

**Comment by LeJimster** [comment]
@allanpfreitas can you put your full log files on here as well?  If they're not too huge you can just drag drop the file into comment.  Rage uses a workaround on mesa, I'm wondering if their is an issue with it on Navi.

**Comment by allanpfreitas** [comment]
@LeJimster sorry but where can I find these logs?

**Comment by LeJimster** [comment]
@allanpfreitas
A few logs:
In the Launch Properties for Rage you can use
`PROTON_LOG=1 %command%` 
It will give you a log file in your home directory called `steam-9200.log`
Also if you look in your SteamLibrary and navigate through `steamapps/compatdata/9200/pfx/drive_c/users/steamuser/Saved Games/id Software/Rage/base/`
you will find some Error logs.

**Comment by allanpfreitas** [comment]
@LeJimster https://gist.github.com/allanpfreitas/e5bb75df70a9c3509e76eb262642f54a

**Comment by LeJimster** [comment]
@allanpfreitas just out of curiosity what happens if you run the game with this launch option:
`allow_glsl_builtin_variable_redeclaration=true %command%`

**Comment by LeJimster** [comment]
> @LeJimster https://gist.github.com/allanpfreitas/e5bb75df70a9c3509e76eb262642f54a

@allanpfreitas Just looking closer at your system I can see elements of the AMDGPU-Pro driver.  Maybe this is causing a conflict?  I would remove the AMDGPU-Pro drivers and stick to the Mesa drivers.

https://linuxconfig.org/amd-radeon-ubuntu-20-04-driver-installation

**Comment by allanpfreitas** [comment]
Yay! Thanks @LeJimster now the game is running but I'm having some textures issues, any tip to solve this one ? 
![Screenshot from 2020-09-30 09-21-52](https://user-images.githubusercontent.com/2490836/94684564-9c2bd880-02fe-11eb-8541-b810181eb7ef.png)
![Screenshot from 2020-09-30 09-22-05](https://user-images.githubusercontent.com/2490836/94684581-9fbf5f80-02fe-11eb-9a1e-d3abf5d3b66a.png)

**Comment by LeJimster** [comment]
@allanpfreitas it's a well known issue as I mentioned in the first post..  The best way is to play the 64-bit version of the game for now.  Others have mentioned using launch options `+vt_maxPPF 128`

**Comment by allanpfreitas** [comment]
Thanks @LeJimster running the 64bit version its works great.

**Comment by Shotweb** [comment]
Running the 64 bit version doesn't solve the corruption issue (like allan's screenshots) for me, neither does seemingly any launch/config options. Using AMD graphics.

**Comment by eukara** [comment]
Same problem on amdgpu as @allanpfreitas. Been trying this on and off since I got this ThinkCentre M75q 2nd Gen in January this year. Running on openSUSE Tumbleweed. I have all sorts of other problems with this type of GPU though. Like freezes in which "drm:amdgpu_cs_ioctl : Failed to initialize parser -125" is spammed into a TTY and my system is fully locked up - with only a force restart able to get me back into action. This happens in a lot of other demanding games too. No idea where to start.

https://gist.github.com/eukara/d1d95c2ac85e273989bfff1cc79516dc < My Steam Hardware report
https://bugzilla.kernel.org/show_bug.cgi?id=205089 Someone else having the exact same freezes as me. I'm wondering as to why amdgpu still suffers from this.

**Comment by kopr12** [comment]
My only problem with amdgpu is with this game, I wonder what's going on with this, I mean is the problem in Mesa or somewhere else, too bad, it's a great game.

**Comment by eukara** [comment]
Using Rage64 and `+vt_maxPPF 128` also doesn't affect anything for me, the game still renders all virtual-texture pages are all over the place. The question obviously becomes whether this is a driver issue, or the game has a bug (and the Windows AMD drivers have a hack to work around this?)
Gotta get this game fixed in time for Steam Deck, which after all runs on AMD! :P

**Comment by zmike** [comment]
For anyone getting the crash on startup from binkyuv, this should be fixed in upcoming versions of mesa.

**Comment by croberts81** [comment]
Finally! Closed issue.
gitlab.freedesktop.org/mesa/mesa/-/issues/1326

**Comment by LeJimster** [comment]
> Finally! Closed issue. gitlab.freedesktop.org/mesa/mesa/-/issues/1326

I need to test it to verify the fix as it's been awhile since I built mesa-git.  Hopefully this does the trick though!
Update:
@croberts81 Yep, just built the latest mesa-git and it is fixed on both 32-bit and 64-bit versions of the game.  No more corrupted tiles!  =)

Big thanks to Timothy Arceri for putting this one finally to bed.  Hopefully this title can be whitelisted now.

**Comment by eukara** [comment]
Thank you all for making this possible! This was my most anticipated game!

**Comment by kopr12** [comment]
Hmm, still broken textures in my case.
`OpenGL version string: 4.6 (Compatibility Profile) Mesa 22.1.3`
`Device: AMD Radeon Vega 3 Graphics (raven, LLVM 14.0.4, DRM 3.42, 5.15.41-gentoo) (0x15dd)`

Tried both 32-bit and 64-bit with several Proton versions.

**Comment by LeJimster** [comment]
> Hmm, still broken textures in my case. `OpenGL version string: 4.6 (Compatibility Profile) Mesa 22.1.3` `Device: AMD Radeon Vega 3 Graphics (raven, LLVM 14.0.4, DRM 3.42, 5.15.41-gentoo) (0x15dd)`
> 
> Tried both 32-bit and 64-bit with several Proton versions.

You need to build the latest mesa-git.  It should appear as Mesa 22.2.0.  There will probably be a stable version in the next few weeks.

**Comment by kopr12** [comment]
> You need to build the latest mesa-git. It should appear as Mesa 22.2.0. There will probably be a stable version in the next few weeks.

Well I checked [release notes](https://docs.mesa3d.org/relnotes/22.1.3.html) for 22.1.3 and it mentioned `[wine] RAGE: texture problems` as `bug fixed`

Anyway, compiling now the latest meta-git and will report back.

**Comment by LeJimster** [comment]
> > You need to build the latest mesa-git. It should appear as Mesa 22.2.0. There will probably be a stable version in the next few weeks.
> 
> Well I checked [release notes](https://docs.mesa3d.org/relnotes/22.1.3.html) for 22.1.3 and it mentioned `[wine] RAGE: texture problems` as `bug fixed`
> 
> Anyway, compiling now the latest meta-git and will report back.

Right.  I will see about testing 22.1.3 on my laptop tomorrow. Let us know if mesa-git works.

**Comment by kopr12** [comment]
Same thing with git version.
`OpenGL version string: 4.6 (Compatibility Profile) Mesa 22.2.0-devel (git-7bb62d9dd8)`

Well I don't know what to say, I tried, maybe it's just me for some odd reason, but I seriously doubt that, I run flatpak version of Steam though, maybe it has something to do with it, but unlikely, since I don't experience problems with other games.

**Comment by LeJimster** [comment]
> Same thing with git version. `OpenGL version string: 4.6 (Compatibility Profile) Mesa 22.2.0-devel (git-7bb62d9dd8)`
> 
> Well I don't know what to say, I tried, maybe it's just me for some odd reason, but I seriously doubt that, I run flatpak version of Steam though, maybe it has something to do with it, but unlikely, since I don't experience problems with other games.

The fix works for me on my PC and laptop.  I just tested it on my laptop against mesa-git as mesa 22.1.3 hasn't been added to Arch stable yet.  I have a Vega 56 in my PC and the laptop is running a 5900HX with integrated Vega graphics.

Laptop:
`OpenGL vendor string: AMD
OpenGL renderer string: RENOIR (renoir, LLVM 14.0.6, DRM 3.46, 5.18.8-arch1-1)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 22.2.0-devel (git-8ab1e9826d)
`

I have never run Steam as a flatpak so I cant comment on whether that would be causing issues.  I always use Steam (runtime).  I'm not really up on flatpaks but could it be that its using an old `00-mesa-defaults.conf` for the flatpak and not applying the new workaround.....?  the workaround requires `force_gl_map_buffer_synchronized` to be enabled.

```
        <application name="RAGE (64-bit)" executable="Rage64.exe">
            <option name="allow_glsl_builtin_variable_redeclaration" value="true" />
            <option name="force_gl_map_buffer_synchronized" value="true" />
        </application>

        <application name="RAGE (32-bit)" executable="Rage.exe">
            <option name="allow_glsl_builtin_variable_redeclaration" value="true" />
            <option name="force_gl_map_buffer_synchronized" value="true" />
```

Try adding `force_gl_map_buffer_synchronized=true %command%` to your launch options see if that helps?

**Comment by kopr12** [comment]
Yeah, that could be the problem, I've updated flatpak version of mesa to `22.1-branchpoint-3487-gb28efd80ebb` , that's the latest one that's available I'm afraid for flatpak, and I think the fix came after that snapshot, so I'll just have to wait it out a little more. Glad that it's finally fixed though, there's time for me to play it :)

**Comment by LeJimster** [comment]
Can confirm the latest mesa stable does indeed fix the virtual texture bug. 
```
OpenGL vendor string: AMD
OpenGL renderer string: AMD RENOIR (LLVM 14.0.6, DRM 3.46, 5.18.9-arch1-1)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 22.1.3
```

**Comment by kopr12** [comment]
Switched from flatpak version to regular native Steam and it works.
Had some issues with 5.1 sound speakers, but I had that with some other Proton games as well, the only Proton version where the sound is working fine is 4.11, but that's another issue I guess.

**Comment by uqs** [comment]
Can confirm that upgrading to `mesa 22.1.7` fixed the texture glitches for me, no command line overrides required anymore. I still experienced sounds not playing or cutting out, and it _might_ be that a combination of `PROTON_NO_ESYNC=1` and ` +jobs_numThreads 2` has fixed this for me, but I need a bit more mileage with it still to see if dialogue sounds or car noises are cutting off.

32bit Steam version, with Proton 7 or something
```
OS: Arch Linux
KERNEL: 5.19.6-arch1-1
CPU: AMD Ryzen 5 5600X 6-Core
GPU: AMD Radeon RX 6800 (sienna_cichlid, LLVM 14.0.6, DRM 3.47, 5.19.6-arch1-1)
GPU DRIVER: 4.6 Mesa 22.1.7
RAM: 32 GB
```

**Comment by kopr12** [comment]
I solved my sound problems with `PULSE_LATENCY_MSEC=60 %command%`
